changelog.d/pr-9621

```diff
@@ -0,0 +1,9 @@
+synopsis: Warn about missing `default-language`
+packages: Cabal cabal-install
+prs: #9621
+issues: #9620
+description: {
+
+- `cabal check` will warn about missing `default-language` and not assume `Haskell 2010`.
```

Comment:
I am not sure this comment is accurate, if you see `Distribution.Simple.GHC.Internal`, not specifying a `default-language` defaults to `Haskell98`.

Reply:
#6926 says otherwise. maybe buildinfo gets populated anyway? we should test with -v3`

Reply:
This is specifically about `check` though.

Would removing “ and not assume `Haskell 2010`” be ok for you?

Reply:
The Cabal changelog also claims it uses the compiler's default language: https://cabal.readthedocs.io/en/stable/file-format-changelog.html#cabal-version-3-4